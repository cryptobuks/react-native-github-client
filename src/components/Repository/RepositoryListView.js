import React, { PureComponent } from 'react';
import { FlatList, StyleSheet, TouchableOpacity } from 'react-native';
import {
  Text,
  List,
  ListItem,
  Grid,
  Row,
  View,
  Input,
  Item,
  Icon,
  ActionSheet,
} from 'native-base';
import Feather from 'react-native-vector-icons/Feather';
import Octicons from 'react-native-vector-icons/Octicons';
import { Mutation } from 'react-apollo';
import type { Node } from 'react';
import _ from 'lodash';

import { addStar, removeStar } from './mutations';
import { openWebView } from 'utils';
import SwipeRow from 'SwipeRow';

type Props = {
  repositories: [Object],
  fetchMore: Node => null,
  refetch: Node => null,
  navigator: Object,
  repoType: ?String,
  login: ?String,
  searchEnabled: ?Boolean,
};

const SORT_OPTIONS = [
  'Default',
  'Name: A - Z',
  'Stargazer Count: High to Low',
  'Fork Count: High to Low',
  'Cancel',
];
const SORT_FUNC = [
  null,
  ({ node }) => _.lowerCase(node.name),
  ({ node }) => -node.stargazers.totalCount,
  ({ node }) => -node.forkCount,
];

/**
 * Display a List of Repositories
 * @extends PureComponent
 */
class RepositoryListView extends PureComponent<Props> {
  static defaultProps = {
    search: false,
  };

  state = {
    refreshing: false,
    filtered: [],
    query: '',
  };
  /**
   * Used in FlatList to make sure that each repo has a distinct key
   * @param  {Object} repo  object containing information for each repo
   * @return {String}      a unique key used to identify each row in list
   */
  repoKeyExtractor = repo => repo.node.id;

  handlePressRepository = repo => () =>
    this.props.navigator.push({
      screen: 'profile.repository',
      title: repo.name,
      passProps: { owner: repo.owner.login, name: repo.name },
    });

  /**
   * Renders the user object into a row in the list
   * @param  {Object} item the object containing information for each repository
   * @return {Node}       a React Native Node that can be rendered into a row
   *                      in the list
   */
  renderRepo = ({ item: { node } }) => (
    <Mutation mutation={addStar}>
      {addStar => (
        <Mutation mutation={removeStar}>
          {removeStar => (
            <View style={node.isPrivate ? styles.privateRepo : null}>
              <SwipeRow
                icon={
                  node.viewerHasStarred
                    ? {
                        name: 'star-border',
                        text: 'Unstar',
                        color: 'firebrick',
                      }
                    : { name: 'star', text: 'Star', color: 'steelblue' }
                }
                onPressButton={() =>
                  (node.viewerHasStarred ? removeStar : addStar)({
                    variables: { starrableId: node.id },
                  })
                }
              >
                <ListItem
                  onPress={this.handlePressRepository(node)}
                  style={styles.listItem}
                >
                  <Grid>
                    <Row>
                      <Octicons name="repo" size={18} />
                      <Text style={styles.repoName} numberOfLines={1}>
                        {this.props.repoType === 'repositories'
                          ? node.name
                          : node.nameWithOwner}
                      </Text>
                    </Row>
                    <Row style={styles.description}>
                      <Text note numberOfLines={3}>
                        {node.description || 'No description'}
                      </Text>
                    </Row>
                    <Row>
                      {!!node.primaryLanguage && (
                        <View style={styles.bottomTag}>
                          <Octicons
                            size={15}
                            name="primitive-dot"
                            color={node.primaryLanguage.color}
                          />
                          <Text style={styles.bottomTagText}>
                            {node.primaryLanguage.name}
                          </Text>
                        </View>
                      )}
                      {!!node.stargazers.totalCount && (
                        <View style={styles.bottomTag}>
                          <Octicons size={15} name="star" />
                          <Text style={styles.bottomTagText}>
                            {node.stargazers.totalCount}
                          </Text>
                        </View>
                      )}
                      {!!node.forkCount && (
                        <View style={styles.bottomTag}>
                          <Octicons size={15} name="repo-forked" />
                          <Text style={styles.bottomTagText}>
                            {node.forkCount}
                          </Text>
                        </View>
                      )}
                    </Row>
                  </Grid>
                </ListItem>
              </SwipeRow>
            </View>
          )}
        </Mutation>
      )}
    </Mutation>
  );

  handleChangeQuery = query => this.setState({ query }, this.filterRepo);

  filterRepo = () =>
    (!!this.state.query || !!this.state.sortBy) &&
    this.setState({
      filtered: _.sortBy(
        this.state.query
          ? _.filter(this.props.repositories, this.repoFilter)
          : this.props.repositories,
        ...[this.state.sortBy]
      ),
    });

  repoFilter = query => ({ node }) =>
    _.includes(
      _.lowerCase(
        this.props.repoType === 'repositories' ? node.name : node.nameWithOwner
      ),
      _.lowerCase(query)
    );

  handlePressFunnel = () =>
    ActionSheet.show(
      {
        options: SORT_OPTIONS,
        title: 'Sorted By...',
        cancelButtonIndex: SORT_OPTIONS.length - 1,
      },
      this.handleSelectOption
    );

  handleSelectOption = buttonIndex =>
    buttonIndex !== SORT_OPTIONS.length - 1 &&
    this.setState({ sortBy: SORT_FUNC[buttonIndex] }, this.filterRepo);

  renderSearchBar = () => (
    <Item style={styles.searchBar}>
      <Icon active name="search" />
      <Input
        placeholder="Begin Searching..."
        onChangeText={this.handleChangeQuery}
        value={this.state.query}
        placeholderTextColor="darkgrey"
      />
      <TouchableOpacity onPress={this.handlePressFunnel}>
        <Feather name="filter" size={23} color="#1780FB" />
      </TouchableOpacity>
    </Item>
  );

  render = () => (
    <List style={styles.listContainer}>
      <FlatList
        data={
          this.props.searchEnabled && (this.state.query || this.state.sortBy)
            ? this.state.filtered
            : this.props.repositories
        }
        renderItem={this.renderRepo}
        ListHeaderComponent={this.props.searchEnabled && this.renderSearchBar}
        ListEmptyComponent={<Text>No Content</Text>}
        keyExtractor={this.repoKeyExtractor}
        onEndReached={this.props.fetchMore(this)}
        onRefresh={this.props.refetch(this)}
        refreshing={this.state.refreshing}
      />
    </List>
  );
}

const styles = StyleSheet.create({
  privateRepo: {
    backgroundColor: '#FFFDF0',
  },
  listItem: { paddingRight: 10 },
  repoName: { fontWeight: 'bold', marginLeft: 5 },
  description: { marginVertical: 10 },
  bottomTag: { flexDirection: 'row' },
  bottomTagText: { fontSize: 14, marginRight: 10, marginLeft: 3 },
  listContainer: {
    flex: 1,
  },
  searchBar: {
    marginLeft: 10,
    marginRight: 10,
  },
});

export default RepositoryListView;
