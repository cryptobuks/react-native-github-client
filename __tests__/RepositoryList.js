import 'react-native';
import React from 'react';
import { ApolloProvider } from 'react-apollo';
import RepositoryListView from '../src/components/Repository/RepositoryListView';

import renderer from 'react-test-renderer';

const repoList = [
  {
    node: {
      id: 'as123oisdf',
      nameWithOwner: 'foo/bar',
      name: 'bar',
      description: '..',
      viewerHasStarred: true,
      primaryLanguage: null,
      url: '',
      isPrivate: false,
      forkCount: 23,
      stargazers: {
        totalCount: 5,
      },
    },
  },
  {
    node: {
      id: 'asdlkf23sdf',
      nameWithOwner: 'segmentation/fault',
      name: 'fault',
      description: 'something to test',
      viewerHasStarred: false,
      primaryLanguage: {
        id: 'dfd2asdf',
        color: '#0000ff',
        name: 'whocares',
      },
      url: '',
      isPrivate: true,
      forkCount: 0,
      stargazers: {
        totalCount: 5,
      },
    },
  },
];

it('repository list view render correctly', () => {
  expect(
    renderer
      .create(
        <ApolloProvider client={{}}>
          <RepositoryListView
            refetch={() => null}
            fetchMore={() => null}
            repositories={repoList}
          />
        </ApolloProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('repository list empty', () => {
  expect(
    renderer
      .create(
        <ApolloProvider client={{}}>
          <RepositoryListView
            refetch={() => null}
            fetchMore={() => null}
            repositories={[]}
          />
        </ApolloProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});
