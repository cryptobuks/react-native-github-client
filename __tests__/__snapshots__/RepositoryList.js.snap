// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repository list view render correctly 1`] = `
<View
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={false}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  rightOpenValue={0}
  style={Object {}}
  swipeToOpenPercent={50}
  virtual={undefined}
>
  <RCTScrollView
    data={
      Array [
        Object {
          "node": Object {
            "description": "..",
            "forkCount": 23,
            "id": "as123oisdf",
            "isPrivate": false,
            "name": "bar",
            "nameWithOwner": "foo/bar",
            "primaryLanguage": null,
            "stargazers": Object {
              "totalCount": 5,
            },
            "url": "",
            "viewerHasStarred": true,
          },
        },
        Object {
          "node": Object {
            "description": "something to test",
            "forkCount": 0,
            "id": "asdlkf23sdf",
            "isPrivate": true,
            "name": "fault",
            "nameWithOwner": "segmentation/fault",
            "primaryLanguage": Object {
              "color": "#0000ff",
              "id": "dfd2asdf",
              "name": "whocares",
            },
            "stargazers": Object {
              "totalCount": 5,
            },
            "url": "",
            "viewerHasStarred": false,
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={null}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={null}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {},
              null,
            ]
          }
          virtual={undefined}
        >
          <View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "firebrick",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 672,
                  "position": "absolute",
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 78,
                    },
                  ],
                  "width": 750,
                  "zIndex": 0,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 10,
                    "width": 78,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 32,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "white",
                        "fontSize": 12,
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Unstar
                </Text>
              </View>
            </View>
            <InteractableView
              animatedValueX={undefined}
              animatedValueY={undefined}
              boundaries={
                Object {
                  "right": 0,
                }
              }
              collapsable={undefined}
              dragEnabled={true}
              dragToss={0.1}
              horizontalOnly={true}
              onAnimatedEvent={[Function]}
              reportOnAnimatedEvents={true}
              snapPoints={
                Array [
                  Object {
                    "damping": 0.6,
                    "tension": 300,
                    "x": 0,
                  },
                  Object {
                    "damping": 0.6,
                    "tension": 300,
                    "x": -78,
                  },
                ]
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 15,
                        "paddingBottom": 13,
                        "paddingRight": 15,
                        "paddingTop": 13,
                      },
                      Object {
                        "paddingRight": 10,
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 18,
                            },
                            undefined,
                            Object {
                              "fontFamily": "Octicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                      >
                        
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            },
                            Object {
                              "fontWeight": "bold",
                              "marginLeft": 5,
                            },
                          ]
                        }
                        uppercase={false}
                        virtual={undefined}
                      >
                        foo/bar
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                          "marginVertical": 10,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        note={true}
                        numberOfLines={3}
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#808080",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "200",
                          }
                        }
                        uppercase={false}
                        virtual={undefined}
                      >
                        ..
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "flexDirection": "row",
                            },
                          ]
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 15,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Octicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 14,
                                "marginLeft": 3,
                                "marginRight": 10,
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          5
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "flexDirection": "row",
                            },
                          ]
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 15,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Octicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 14,
                                "marginLeft": 3,
                                "marginRight": 10,
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          23
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </InteractableView>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "#FFFDF0",
              },
            ]
          }
          virtual={undefined}
        >
          <View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "steelblue",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 672,
                  "position": "absolute",
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 78,
                    },
                  ],
                  "width": 750,
                  "zIndex": 0,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 10,
                    "width": 78,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 32,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "white",
                        "fontSize": 12,
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Star
                </Text>
              </View>
            </View>
            <InteractableView
              animatedValueX={undefined}
              animatedValueY={undefined}
              boundaries={
                Object {
                  "right": 0,
                }
              }
              collapsable={undefined}
              dragEnabled={true}
              dragToss={0.1}
              horizontalOnly={true}
              onAnimatedEvent={[Function]}
              reportOnAnimatedEvents={true}
              snapPoints={
                Array [
                  Object {
                    "damping": 0.6,
                    "tension": 300,
                    "x": 0,
                  },
                  Object {
                    "damping": 0.6,
                    "tension": 300,
                    "x": -78,
                  },
                ]
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 15,
                        "paddingBottom": 13,
                        "paddingRight": 15,
                        "paddingTop": 13,
                      },
                      Object {
                        "paddingRight": 10,
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 18,
                            },
                            undefined,
                            Object {
                              "fontFamily": "Octicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                      >
                        
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            },
                            Object {
                              "fontWeight": "bold",
                              "marginLeft": 5,
                            },
                          ]
                        }
                        uppercase={false}
                        virtual={undefined}
                      >
                        segmentation/fault
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                          "marginVertical": 10,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        note={true}
                        numberOfLines={3}
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#808080",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "200",
                          }
                        }
                        uppercase={false}
                        virtual={undefined}
                      >
                        something to test
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "flexDirection": "row",
                            },
                          ]
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#0000ff",
                                "fontSize": 15,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Octicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 14,
                                "marginLeft": 3,
                                "marginRight": 10,
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          whocares
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "flexDirection": "row",
                            },
                          ]
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 15,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Octicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 14,
                                "marginLeft": 3,
                                "marginRight": 10,
                              },
                            ]
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                          5
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </InteractableView>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
